src/alternating_methods.jl

```diff
@@ -74,21 +74,12 @@ function alternating_linear_minimization(
             for i in 1:N
                 grad!(gradf.blocks[i], x.blocks[i])
             end
-            t = [2.0 * (N * b - sum(x.blocks)) for b in x.blocks]
+            t = [N * b - sum(x.blocks) for b in x.blocks]
             return storage.blocks = λ[] * gradf.blocks + t
         end
     end
 
-    function dist2(x::BlockVector) 
-        s = 0
-        for i=1:N
-            for j=1:i-1
-                diff = x.blocks[i] - x.blocks[j]
-                s += fast_dot(diff, diff)
-            end
-        end
-        return s
-    end
+    dist2(x::BlockVector) = 0.5 * sum(fast_dot(x.blocks[i]-x.blocks[j], x.blocks[i]-x.blocks[j]) for i in 1:N for j in 1:i-1)
```

Comment:
maybe check how costly this is, given that we are allocating a difference of two vectors for each block